fetch/client/src/main/scala/endpoints4s/fetch/Endpoints.scala

```diff
@@ -318,7 +322,7 @@ trait EndpointsWithCustomErrors
           .orElse(maybeClientErrors)
           .orElse(maybeServerError) match {
           case None =>
-            onload(Left(new Exception(s"Unexpected response status: ${fetchResponse.status}")))
+            onerror(new Exception(s"Unexpected response status: ${fetchResponse.status}"))
```

Comment:
That seems to be an important semantic change. Is this intended?

Reply:
Another leftover from `FutureLike`. It probably would have been a good idea to simply rollback `FutureLike` changes with Git instead of manually removing them 😅 

I will roll this back too for uniformity with XHR, but both `onerrror` and `onload(Left)` end up as `failure` in concrete interpreters for `Future` and `Promise`, so maybe the current callback abstractions are too powerful for the way they end up being used.